### PR TITLE
[IMP] mail_plugin: hide the create button if we do not have access right

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
Hide the "Create Partner" button if we do not have the "create" access
right.

Do not offer to create a project if the current user can't.

Task-2826471

### CODE REVIEW

File: addons/mail_plugin/controllers/mail_plugin.py [hunk 1]
```diff
@@ -366,6 +370,18 @@ def _get_partner_data(self, partner):
         partner_values['title'] = partner.function
         partner_values['enrichment_info'] = None
 
+        if not partner_values['name']:
+            # Always ensure that the partner has a name
+            partner_values['name'] = partner_values['email']
```
Comment (tde-banana-odoo) [inline addons/mail_plugin/controllers/mail_plugin.py:375]: Maybe we could use _parse_partner_name on email and try to find a name and an email ?

File: addons/mail_plugin/controllers/mail_plugin.py [hunk 2]
```diff
@@ -366,6 +370,18 @@ def _get_partner_data(self, partner):
...
+
+        can_write_on_partner = partner.check_access_rights('write', raise_exception=False)
```
Comment (tde-banana-odoo) [inline addons/mail_plugin/controllers/mail_plugin.py:377]: Can't we put both inside the try/except ? Would simplify a bit the code :)

File: addons/mail_plugin/controllers/mail_plugin.py [hunk 3]
```diff
@@ -387,7 +403,9 @@ def _get_contact_data(self, partner):
 
         return {
             'partner': partner_response,
-            'user_companies': request.env['res.users'].browse(request.uid).company_ids.ids
+            'user_companies': request.env['res.users'].browse(request.uid).company_ids.ids,
```
Comment (tde-banana-odoo) [inline addons/mail_plugin/controllers/mail_plugin.py:406]: Don't we have request.env.user ?

**Review by tde-banana-odoo — COMMENTED** [review]
Zbouik !

**Review by std-odoo — COMMENTED** [review]
@tde-banana-odoo Changes done :)

File: addons/project_mail_plugin/static/src/to_translate/translations_outlook.xml [hunk 4]
```diff
@@ -9,4 +9,6 @@
     <string>Create %(name)s</string>
     <string>Search Projects...</string>
     <string>Log Email Into Task</string>
+    <string>The Contact needs to exist to create Task.</string>
+    <string>No Project Found</string>
```
Comment (tde-banana-odoo) [inline addons/project_mail_plugin/static/src/to_translate/translations_outlook.xml:13]: No project or No Project Found ?

Comment (std-odoo) [reply addons/project_mail_plugin/static/src/to_translate/translations_outlook.xml:13]: both should be added, good catch

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**